src/config/types.ts

```diff
@@ -1,19 +1,40 @@
-import { UnderlyingCollateralToken } from '../types';
+import { Provider } from 'ethers';
+import { Decimal } from '@tempusfinance/decimal';
+import { Token, UnderlyingCollateralToken } from '../types';
 
 export type SupportedNetwork = 'goerli' | 'mainnet';
 
-type TickerToAddressMap = {
+type UnderlyingTokenTickerToCollateralAddressMap = {
   [tokenTicker in UnderlyingCollateralToken]: string;
 };
 
+type UnderlyingTokenTickerToDebtAddressMap = {
+  [tokenTicker in UnderlyingCollateralToken]: string;
+};
+
+type TokenConfig = {
+  address: string;
+  ticker: Token;
+  supportsPermit: boolean;
+  positionManager: string;
+  positionManagerABI: string;
+  positionMangerManageFunc: string;
+  priceFeedTicker: UnderlyingCollateralToken | null;
+  hardcodedPrice: Decimal | null;
+  underlyingTokenTicker: UnderlyingCollateralToken | null;
+  underlyingCollateralRate: Decimal | ((address: string, provider: Provider) => Promise<Decimal>) | null;
+};
+
+export type TokenTickerToTokenConfigMap = {
+  [tokenTicker in Token]: TokenConfig;
+};
+
 export interface NetworkConfig {
-  wstEth: string;
-  stEth: string;
-  r: string;
-  raftCollateralTokens: TickerToAddressMap;
-  raftDebtToken: string;
+  raftCollateralTokens: UnderlyingTokenTickerToCollateralAddressMap;
+  raftDebtTokens: UnderlyingTokenTickerToDebtAddressMap;
   positionManager: string;
   positionManagerStEth: string;
-  priceFeeds: TickerToAddressMap;
+  priceFeeds: UnderlyingTokenTickerToCollateralAddressMap;
```

Comment:
Either add `UnderlyingTokenTickerToPriceFeedAddressMap` or just revert `TickerToAddressMap` as a generic type and use it for all three.